src/dune_rules/compilation_context.mli

```diff
@@ -27,6 +27,7 @@ val create
   -> flags:Ocaml_flags.t
   -> requires_compile:Lib.t list Resolve.Memo.t
   -> requires_link:Lib.t list Resolve.t Memo.Lazy.t
+  -> ?implements_parameter:Module_name.t * Module_name.t option Resolve.Memo.t
```

Comment:
Can you turn this tuple into a record? I found myself back tracking to determine these two fields a few too many times.